wasmcloud-chat/actors/messages/src/cloudevent.rs

```diff
@@ -0,0 +1,46 @@
+use chrono::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Repesents a cloudevent v1.0 JSON format message.
+/// for ease of use, copy this struct into any library that needs
+/// to round-trip these structures. Don't create the unnecessary headache
+/// of creating coupling just for the use of cloud events.
+#[derive(Serialize, Deserialize, PartialEq, Clone, Default)]
+pub struct CloudEvent {
+    #[serde(rename = "specversion")]
+    pub cloud_events_version: String,
+    #[serde(rename = "type")]
+    pub event_type: String,
+    pub source: String, // URI
+    #[serde(rename = "id")]
+    pub event_id: String,
+    #[serde(rename = "time")]
+    pub event_time: String,
+    #[serde(rename = "datacontenttype")]
+    pub content_type: String,
+    pub data: serde_json::Value,
+}
```

Comment:
If other libraries may need to refer to this struct, could we put this into the messaging interface? I figure that's what your coupling comment is about, no need to tie X library to the interface just for the cloud event?

Reply:
yeah, we want to avoid creating a real dependency between libraries until it is no longer avoidable. This JSON structure should not be a reason why we tie crates together.

Reply:
I'm fine with leaving this here for now, I think the only downside would be needing to change this struct in multiple places if it's used elsewhere. If `CloudEvent` is being used all over the place maybe we can centralize it

Reply:
We'd also lose a bit of fidelity by putting this struct into the messaging interface. Firstly, it'd be hard to get the names in the `widl` to come out looking properly in the JSON fields, and secondly the `data` field here is of type `serde_json::Value`, which is subtly different than making it a string. In the latter, we have to escape all the quotes in the nested JSON content, in the former we do not.

Reply:
Put another way, other participants in the exchange of "cloud events" have varying concerns. Some might just want to grab the whole thing as a single opaque JSON payload, while others might be interested in just the envelope. So, rather than having them all create this tight coupling on this one Rust struct, we're just letting everyone apply "just enough" strong typing to the JSON payload as they need.

Reply:
We shouldn't have to change the `CloudEvent` struct given that it treats the `data` field as opaque. If this JSON changes, it's because the core Cloud Events specification changed, and there's even a "spec version" field on this JSON so that consumers _not_ relying on this specific Rust structure can adapt to future changes.